test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java

```diff
@@ -467,6 +508,27 @@ static List<String> extractPartEtags(BytesReference completeMultipartUploadBody)
         }
     }
 
+    private static final Pattern rangePattern = Pattern.compile("^bytes=([0-9]+)-([0-9]+)$");
+
+    private static Tuple<Long, Long> parsePartRange(final HttpExchange exchange) {
```

Comment:
Could we use `org.elasticsearch.test.fixture.HttpHeaderParser#parseRangeHeader` here? Although I do like how this pattern is anchored at start and end, maybe we should do that in `HttpHeaderParser` too.

Reply:
Ah sure, I hadn't seen it, will do. I am inclined not to change that pattern here but happy to do it as a self-contained change.